include/engine/guidance/route_step.hpp

```diff
@@ -71,55 +71,117 @@ struct RouteStep
     // indices into the locations array stored the LegGeometry
     std::size_t geometry_begin;
     std::size_t geometry_end;
-    std::vector<Intersection> intersections;
+    std::vector<IntermediateIntersection> intersections;
 
-    LaneID numLanesToTheRight() const
+    // remove all information from the route step, marking it as invalid (used to indicate empty
+    // steps to be removed).
+    void Invalidate()
+    {
+        name_id = EMPTY_NAMEID;
+        name.clear();
+        ref.clear();
+        pronunciation.clear();
+        destinations.clear();
+        rotary_name.clear();
+        rotary_pronunciation.clear();
+        duration = 0;
+        distance = 0;
+        mode = TRAVEL_MODE_INACCESSIBLE;
+        maneuver = getInvalidStepManeuver();
+        geometry_begin = 0;
+        geometry_end = 0;
+        intersections.clear();
+        intersections.push_back(getInvalidIntersection());
+    }
+
+    // Elongate by another step in front
+    RouteStep &MakePreceededBy(const RouteStep &preceeding_step)
```

Comment:
Hm the name is a bit confusing to me to be honest.

Especially because this is `Make` prefixed and elongate (or others) below are not. They return a Step& too.